src/pusb.c

```diff
@@ -359,11 +359,13 @@ int send_file(libusb_device_handle *udev, char *f_name)
 			i++;
 		}
 	}
+	usb_sleep(50);
```

Comment:
What is the purpose of delay - how do we know the delays are valid and not hiding some other symptom?

Reply:
Hmm, I said everything I know in the commit message for that particular part, but I will try to elaborate more.

The omap seems to need some delay between commands to successfully
process them.
Symptoms were: failed ASIC reads when verbosity is low
(so not much delay by printing stuff out), failed sendfile at first
block.

I checked in wireshark what is going on on the usb bus and I did not see anything suspicious besides of failed out transfers. pusb does not look suspicious for internal timing problems, no multithreading, e.g. 

Well I am quite sure these delays should not harm.
So the only suspicion I have is that the omap cannot handle new commands at certain points.
But since I do not have the source code of that boot rom, I cannot find out what is really going on there. I can only know what happens on the usb bus.

Reply:
maybe older hardware had enough intrinsic delays

Reply:
Understood, please add a code comment to the effect for someone in the future to understand. Thanks for digging.